src/controller/SoldierController.ts

```diff
@@ -0,0 +1,37 @@
+import { Request, Response } from 'express';
+import * as z from 'zod';
+import soldierSchema from '../repository/schemaValidation';
+import logger from '../logger';
+import { createSoldier, getSoldiers, getSoldiersByQuery } from '../repository/soldierRepo';
+
+export async function insertSoldier(req:Request, res:Response) {
+  try {
+    soldierSchema.parse(req.body);
+    const savedSoldier = await createSoldier(req.body);
+    logger.info(savedSoldier);
+    res.status(201).send(savedSoldier);
+  } catch (err) {
+    if (err instanceof z.ZodError) {
+      res.sendStatus(400);
+    } else res.sendStatus(400);
+  }
+}
+
+export async function getSoldierByID(req:Request, res:Response) {
+  const { id } = req.params;
```

Comment:
parse the req params for id

Because you are doing parsing at the beginning of every route, we use middlewares for this job.

Look up express - middlewares.


Do so in the next task